packages/std/src/exports.rs

```diff
@@ -1,5 +1,3 @@
-#![cfg(target_arch = "wasm32")]
```

Comment:
Why remove those?
They seem like nice safety (I do agree with adding the guard in `lib.rs` as well, but I prefer to have these non-importable by any means if we are not targetting wasm

Reply:
Moving the guard to `lib.rs` avoids the need of a comment and importing an empty module. So far we are on the same page.

If I understand correctly, it is no possible to import relative paths on Rust. Everything needs to be relative to `lib.rs`/`crate::`, so if I understand correctly it is not possible to import them accidentally.

Reply:
Yes, you must explicitly import them `crate::exports::XYZ`. Just that no one does that from another file. But then again, we are the only ones working on the crate, so this is probably just paranoia to add more checks and guards

Reply:
Even this way, the code does not compile on non-wasm32, since `crate::exports` is not available.

This diff

```diff
diff --git a/packages/std/src/demo.rs b/packages/std/src/demo.rs
index 8284fec..7717955 100644
--- a/packages/std/src/demo.rs
+++ b/packages/std/src/demo.rs
@@ -4,9 +4,11 @@
 #![allow(dead_code)]
 
 use crate::traits::{ReadonlyStorage, Storage};
+use crate::exports::{allocate};
 
 // returns the length as a 2 byte big endian encoded integer
 fn len(prefix: &[u8]) -> [u8; 2] {
+    let _ = allocate(3);
     if prefix.len() > 0xFFFF {
         panic!("only supports namespaces up to length 0xFFFF")
     }
```

compiles with `cargo wasm` but fails when trying to `cargo build` because:

```
cargo build           
   Compiling cosmwasm v0.7.2 (/projects/cosmwasm/packages/std)
error[E0432]: unresolved import `crate::exports`
 --> packages/std/src/demo.rs:7:12
  |
7 | use crate::exports::{allocate};
  |            ^^^^^^^ maybe a missing crate `exports`?

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `cosmwasm`.
```

Reply:
Okay, then ignore my comment... excessive compiler paranoia :laughing:

Reply:
I'm glad we talked about it :) Interesting stuff to explore here.